pkg/kratos/handlers.go

```diff
@@ -753,6 +775,30 @@ func (a *API) handleUpdateSettingsFlow(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// populate returnTo url if not set
+	if flow.ReturnTo == nil {
+		if continueWith, ok := flow.GetContinueWithOk(); ok {
+			for _, c := range continueWith {
+				if continueRedirect := c.ContinueWithRedirectBrowserTo; continueRedirect != nil {
+					flow.SetReturnTo(continueRedirect.RedirectBrowserTo)
+					break
+				}
+			}
+		}
+	}
+
+	// force redirection after successful webauthn registration
+	if _, ok := body.GetActualInstance().(*client.UpdateSettingsFlowWithWebAuthnMethod); ok {
+		if state, ok := flow.State.(string); ok && state == "success" {
+			if returnTo, ok := flow.GetReturnToOk(); ok {
+				a.redirectResponse(w, r, &BrowserLocationChangeRequired{
+					RedirectTo: returnTo,
+				})
+				return
+			}
+		}
+	}
```

Comment:
question: can you add a comment about why whit is needed? I am not sure I understand what this does, wouldn't the frontend take care of sending the user to the return_to?